migrations/manual/educandu-2021-12-14-01-migrate-cdn-resources-once-more.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable camelcase, no-await-in-loop */
+
+const pluginsWithCdnResources = new Set([
+  'image',
+  'video',
+  'image-tiles',
+  'audio',
+  'anavis',
+  'ear-training',
+  'markdown'
+]);
+
+const INTERNAL_URL_TYPE = 'internal';
+const processImageSection = ({ content }) => {
+  const urls = content?.sourceType === INTERNAL_URL_TYPE ? [content.sourceUrl] : [];
+
+  if (content?.effect?.sourceType === INTERNAL_URL_TYPE) {
+    urls.push(content.effect.sourceUrl);
+  }
+
+  return urls;
+};
+
+const processVideoSection = ({ content }) => content?.sourceType === INTERNAL_URL_TYPE ? [content.sourceUrl] : [];
+
+const processImageTilesSection = ({ content }) => {
+  return content?.tiles?.reduce((acc, tile) => {
+    if (tile.image?.type === INTERNAL_URL_TYPE) {
+      acc.push(tile.image.url);
+    }
+
+    return acc;
+  }, []) || [];
+};
+
+const processEarTrainingSection = ({ content }) => {
+  return content?.tests?.reduce((acc, test) => {
+    if (test.sound?.type === INTERNAL_URL_TYPE) {
+      acc.push(test.sound.url);
+    }
+
+    return acc;
+  }, []) || [];
+};
+
+const processAnavisSection = ({ content }) => content?.media?.type === INTERNAL_URL_TYPE ? [content.media.url] : [];
+
+const processAudioSection = ({ content }) => content?.type === INTERNAL_URL_TYPE ? [content.url] : [];
+
+const processMarkdownSection = ({ content }) => {
+  if (!content) {
+    return content;
+  }
+  const regexp = /(?<=\]\(cdn:\/\/)(.*?)(?=\))/g;
+  const matches = content.match(regexp);
+  return matches ? [...new Set(matches)] : [];
+};
+
+export const aggregateSectionUrls = sections => sections.reduce((acc, section) => {
+  switch (section.type) {
+    case 'image':
+      return [...acc, ...processImageSection(section)];
+    case 'video':
+      return [...acc, ...processVideoSection(section)];
+    case 'image-tiles':
+      return [...acc, ...processImageTilesSection(section)];
+    case 'ear-training':
+      return [...acc, ...processEarTrainingSection(section)];
+    case 'anavis':
+      return [...acc, ...processAnavisSection(section)];
+    case 'audio':
+      return [...acc, ...processAudioSection(section)];
+    case 'markdown':
+      return [...acc, ...processMarkdownSection(section)];
```

Comment:
This is the same migration script content as in `educandu-2021-11-18-02-migrate-cdn-resoures.js`, except for the 'markdown' case which is new.